repair/row_level.cc

```diff
@@ -707,6 +737,11 @@ class repair_meta {
     // sharding info of repair master
     dht::sharder _remote_sharder;
     bool _same_sharding_config = false;
+    struct local_range_estimation {
+        size_t master_subranges_count;
+        size_t partitions_count;
+    };
```

Comment:
Counting partitions is wrong, as partitions can be very large.

Better to count bytes, we have byte counts for sstables and memtables already.

We should consider the number of bytes that are going to be discarded. Say, if we're going to discard more than 1MB, then better to use shard hopping. I guess the number of bytes to be discarded =~ total number of bytes in the table, so it's enough to check if the table size is smaller than a megabyte.

Reply:
> I guess the number of bytes to be discarded =~ total number of bytes in the table, so it's enough to check if the table size is smaller than a megabyte.

We can have a small sized table with large number of partitions, `multishard_filter` will have a lot of overhead from `next_partition` calls in this case. The heuristic with `master_subranges_count ` and `partitions_count` tries to minimise the number of hops, either `next_partition` for  `multishard_filter` or `fast_forward_to` for `multishard_split`, leaving discarded bytes overhead aside. This is elaborated in [this comment](https://github.com/scylladb/scylladb/pull/14178/files#diff-e8909a2855e30487ffe474441ada239cfe8401fe46b51d0b28718fac529d82dbR1101).